lib/dialects/mysql/query-generator.js

```diff
@@ -151,6 +155,132 @@ const QueryGenerator = {
     return `ALTER TABLE ${this.quoteTable(tableName)} CHANGE ${attrString.join(', ')};`;
   },
 
+  // A recursive parser for nested where conditions
+  parseConditionObject(_conditions, path) {
+    path = path || [];
+    return Utils._.reduce(_conditions, (r, v, k) => { // result, key, value
+      if (Utils._.isObject(v)) {
+        r = r.concat(this.parseConditionObject(v, path.concat(k))); // Recursively parse objects
+      } else {
+        r.push({ path: path.concat(k), value: v });
+      }
+      return r;
+    }, []);
+  },
+
+  handleSequelizeMethod(smth, tableName, factory, options, prepend) {
+    if (smth instanceof Utils.Json) {
+      // Parse nested object
+      if (smth.conditions) {
+        const conditions = _.map(this.parseConditionObject(smth.conditions), condition =>
+          `${this.quoteIdentifier(_.first(condition.path))}->>'\$.${_.tail(condition.path).join('.')}' = '${condition.value}'`
+        );
+
+        return conditions.join(' and ');
+      } else if (smth.path) {
+        let str;
+
+        // Allow specifying conditions using the mysql json syntax
+        if (_.some(['->', '->>'], _.partial(_.includes, smth.path))) {
+          str = smth.path;
+        } else {
+          // Also support json dot notation
+          const path = smth.path.split('.');
+          str = `${this.quoteIdentifier(_.first(path))}->>'\$.${_.tail(path).join(',')}'`;
+        }
+
+        if (smth.value) {
+          str += util.format(' = %s', this.escape(smth.value));
+        }
+
+        return str;
+      }
+    } else {
+      return AbstractQueryGenerator.handleSequelizeMethod.call(this, smth, tableName, factory, options, prepend);
+    }
+  },
+
+  whereItemQuery(key, value, options) {
+    options = options || {};
+
+    let field = options.field || options.model && options.model.rawAttributes && options.model.rawAttributes[key] || options.model && options.model.fieldRawAttributesMap && options.model.fieldRawAttributesMap[key];
+    let fieldType = options.type || (field && field.type);
+
+    if (key && typeof key === 'string' && key.indexOf('.') !== -1 && options.model) {
+      if (options.model.rawAttributes[key.split('.')[0]] && options.model.rawAttributes[key.split('.')[0]].type instanceof DataTypes.JSON) {
+        field = options.model.rawAttributes[key.split('.')[0]];
+        fieldType = field.type;
+        const tmp = value;
+        value = {};
+
+        Dottie.set(value, key.split('.').slice(1), tmp);
+        key = field.field || key.split('.')[0];
+      }
+    }
+
+    if (_.isPlainObject(value) && fieldType instanceof DataTypes.JSON) {
```

Comment:
No reason in particular, I inspired myself from the [AbstractQueryGenerator](https://github.com/rundef/sequelize/blob/mysql_json_datatype/lib/dialects/abstract/query-generator.js#L1938). I'll remove it.